### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by himanshukodwani** [zzzprojects/EntityFramework-Plus#596]
1) Is there a way to know which methods are part of the paid version and which methods are part of community version.
2) Is there a way to expire QueryCache manually based on entity type or whenever a Create/Update/Delete operation takes place on the entity when such an entity is part of a query cache.

**Comment by JonathanMagnan** [comment]
Hello @himanshukodwani ,

> Is there a way to know which methods are part of the paid version and which methods are part of community version.

At the bottom of the download section, you can found them: https://entityframework-plus.net/download

In short, here are the following method that are paid:
- BulkSaveChanges
- BatchSaveChanges
- BulkInsert
- BulkUpdate
- BulkDelete
- BulkMerge
- BulkSynchronize

Everything else should be free.

--

> Is there a way to expire QueryCache manually based on entity type or whenever a Create/Update/Delete operation takes place on the entity when such an entity is part of a query cache.

About the entity type, I don't think. However, you can manage it on your side with `Tag` and using `QueryCacheManager.ExpireTag` method. It could probably be possible on our side to add something about the entity type if you really need it.

Are you using `EF6`? If so, that's possible via `QueryCacheManager.IsAutoExpireCacheEnabled = true`. Whenever a `DbSet` is modified, it will automatically expire all items related to it.

If you are using `EF Core`, the `IsAutoExpireCacheEnabled` is not yet available but we are currently reviewing the model code so perhaps it will be eventually possible.

---

Let me know if that answer correctly to your both question.

Best Regards,

Jon

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by himanshukodwani** [comment]
Hi Jon,
Thanks for your prompt reply.
Noted all. By the way I am using EF core and I think feature like QueryCacheManager.IsAutoExpireCacheEnabled = true is the one I need.
Thanks and best regards,
Himanshu

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by JonathanMagnan** [comment]
> QueryCacheManager.IsAutoExpireCacheEnabled = true is the one I need.

Yup, I said, it's not yet available for .NET Core (when we tried, we failed) but we will look again once we complete our current model development.
